File: 107598038/HW2/src/main/src/presenter/editCoursePresenter.java
package presenter;

import error.CourseError;
import output.editCourseOutput;
import view.ViewModel;

public class editCoursePresenter implements Presenter, editCourseOutput {
    CourseError error = new CourseError();
    @Override
    public ViewModel createView() {
        return null;
    }

    @Override
    public CourseError getCourseError() {
        return error;
    }
}

File: 107598019/HW2/Software Architecture_MySQL/Software Architecture/adapter/presenter/EditCoursePresenter.cs
using Software_Architecture.usecase.editCourse;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Software_Architecture.adapter.presenter
{
    public class EditCoursePresenter : EditCourseOutput
    {
        public EditCoursePresenter() { }

        private string newTitle;
        private ViewModel vm = new ViewModel();

        public string getNewTitle()
        {
            return newTitle;
        }

        public void setNewTitle(string newTitle)
        {
            this.newTitle = newTitle;
            vm.setTitle(newTitle);
        }

        public ViewModel getViewModel()
        {
            return vm;
        }
    }
}
